src/linker/Linker.Steps/LinkAttributesParser.cs

```diff
@@ -117,16 +125,22 @@ static string FormatCustomAttribute (CustomAttribute ca)
 			//		public RemoveAttributeInstancesAttribute (object value1) {}
 			// }
```

Comment:
I think this is worth updating the comment and adding details to it. Specifically:
* We generate new constructors for the syntetic `RemoveAttributeInstancesAttribute` type (in the Cecil space, with no namespace) based on the number of arguments in the XML
* Each such .ctor is defined as `.ctor(object, object, ....)` - all arguments are boxed then in the XML by wrapping them with "object" argument.
* The internal linker representation which is a type called `Mono.Linker.RemoveAttributeInstancesAttribute` has only one `.ctor` which takes a collection of arguments. The linker attribute caching mechanism (link) converts the call to `.ctor` with arbitrary number of arguments to the instantiation of the internal representation.
* The boxing is removed in this conversion

Reply:
Note, I'd keep the default ctor